src/App/Pages/Send/SendAddEditPageViewModel.cs

```diff
@@ -0,0 +1,512 @@
+                }
+            }
+        }
+        public int? MaxAccessCount
+        {
+            get => _maxAccessCount;
+            set
+            {
+                if (SetProperty(ref _maxAccessCount, value))
+                {
+                    MaxAccessCountChanged();
+                }
+            }
+        }
+        public SendView Send
+        {
+            get => _send;
+            set => SetProperty(ref _send, value, additionalPropertyNames: _additionalSendProperties);
+        }
+        public bool ShowEditorSeparators
+        {
+            get => _showEditorSeparators;
+            set => SetProperty(ref _showEditorSeparators, value);
+        }
+        public Thickness EditorMargins
+        {
+            get => _editorMargins;
+            set => SetProperty(ref _editorMargins, value);
+        }
+        public bool ShowPassword
+        {
+            get => _showPassword;
+            set => SetProperty(ref _showPassword, value,
+                additionalPropertyNames: new string[]
+                {
+                    nameof(ShowPasswordIcon)
+                });
+        }
+        public bool EditMode => !string.IsNullOrWhiteSpace(SendId);
+        public bool IsText => Send?.Type == SendType.Text;
+        public bool IsFile => Send?.Type == SendType.File;
+        public bool ShowDeletionCustomPickers => EditMode || DeletionDateTypeSelectedIndex == 6;
+        public bool ShowExpirationCustomPickers => EditMode || ExpirationDateTypeSelectedIndex == 7;
+        public string ShowPasswordIcon => ShowPassword ? "" : "";
+
+        public void Init()
+        {
+            PageTitle = EditMode ? AppResources.EditSend : AppResources.AddSend;
+        }
+
+        public async Task<bool> LoadAsync()
+        {
+            var userService = ServiceContainer.Resolve<IUserService>("userService");
+            _canAccessPremium = await userService.CanAccessPremiumAsync();
+            // TODO Policy Check
+            if (Send == null)
+            {
+                if (EditMode)
+                {
+                    var send = await _sendService.GetAsync(SendId);
+                    if (send == null)
+                    {
+                        return false;
+                    }
+                    Send = await send.DecryptAsync();
+                }
+                else
+                {
+                    Send = new SendView
+                    {
+                        Type = Type.GetValueOrDefault(SendType.Text),
+                        DeletionDate = DateTime.Now.AddDays(7),
+                    };
+                    DeletionDateTypeSelectedIndex = 4;
+                    ExpirationDateTypeSelectedIndex = 0;
+                }
+
+                TypeSelectedIndex = TypeOptions.FindIndex(k => k.Value == Send.Type);
+                MaxAccessCount = Send.MaxAccessCount;
+                _isOverridingPickers = true;
+                DeletionDate = Send.DeletionDate.ToLocalTime();
```

Comment:
I'll go ahead and make it explicit.  I think you may be right about `"d"` though I can't find any documented proof, aside from the date/time results working as expected.  :)  I do explicitly set the expiration date to midnight the first time expiration time is set without a date (line 498).

Reply:
After some thought I ended up performing the "midnighting" during the calculation phase in `UpdateSendData()` so it will cover for any shenanigans that may happen after the initial load.  I'm also iffy on messing with the time on the way into the DatePicker as the display conversion from UTC to local could be messed up if/when crossing over midnight.  The picker is currently handling that gracefully for us and I appreciate that magic.